debug_util.go

```diff
@@ -0,0 +1,66 @@
+package s2db_arrow_driver
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+)
+
+func logQueryExecution(loggingEnabled bool, query string, args ...interface{}) {
+	if loggingEnabled {
+		fmt.Printf("Executing query (query: '%s', args: '%v')\n", query, args)
+	}
+}
+
+func profileQuery(loggingEnabled bool, ctx context.Context, conn *sql.Conn, query string) {
+	if !loggingEnabled {
+		return
+	}
+
+	execute := func(query string) error {
+		logQueryExecution(loggingEnabled, query)
+		_, err := conn.ExecContext(ctx, query)
+		if err != nil {
+			fmt.Printf("Failed to perform profiling (%s)\n", err)
+			return err
+		}
+
+		return nil
+	}
```

Comment:
I think it makes sense to create a global utility function like this. It should combine `logQueryExecution` and `conn.ExecContext`. Another version would combine `logQueryExecution` and `conn.QueryContext`.